### PR TITLE
isc: use layout access plan API

### PR DESCRIPTION
**Pull request by andriytk**
Use Motr [layout access plan API](https://github.com/Seagate/cortx-motr/blob/main/layout/plan.h) to access the data at the server side for the computation.

This PR depends on https://github.com/Seagate/cortx-motr/pull/606.

### CODE REVIEW

**Review by siningwuseagate — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -193,7 +193,7 @@ clean: isc-clean
 	rm -f $(FGEN)
 	rm -f sfilet-* snodet-* fidout-*
 	rm -f upFile dwFile
-	rm -f *.o *.d *.d.* help.h
+	rm -f *.o *.d *.d.*
```
Comment (nikitadanilov) [inline Makefile:196]: Just for your information: the licencing header at the top of this file is outdated.

File: c0isc_demo.c [hunk 2]
```diff
@@ -245,63 +232,174 @@ static int ping_input_prepare(struct m0_buf *buf, struct m0_fid *comp_fid,
...
+	if (rc < 1) {
+		fprintf(stderr, "failed to read the resulting xr-value\n");
+		m0_free(buf);
+		return NULL;
+	}
+
+	y->mr_idx     += x->mr_idx_max;
+	y->mr_idx_max += x->mr_idx_max;
+
+	//printf("buf=%s x_rval=%lf\n", buf, x_rval);
```
Comment (nikitadanilov) [inline c0isc_demo.c:281]: C, please. The same below.

File: isc/libdemo.h [hunk 3]
```diff
@@ -22,19 +22,47 @@
...
+#include "fid/fid_xc.h"
+#include "lib/vec.h"
+#include "lib/vec_xc.h"
 #include "xcode/xcode_attr.h"
 
-/** Holds the result of min and max computations. */
+struct isc_buf {
+	uint32_t  i_len;
+	char     *i_buf;
+} M0_XCA_SEQUENCE;
```
Comment (nikitadanilov) [inline isc/libdemo.h:34]: Why not re-use m0_buf?

File: isc/libdemo.h [hunk 4]
```diff
@@ -22,19 +22,47 @@
...
+	struct isc_buf mr_lbuf;
+	/** left  cut of the value on the right side of unit */
+	struct isc_buf mr_rbuf;
 } M0_XCA_RECORD;
 
-struct isc_args {
+struct isc_arr {
 	uint32_t ia_len;
 	double  *ia_arr;
 } M0_XCA_SEQUENCE;
```
Comment (nikitadanilov) [inline isc/libdemo.h:57]: Same question.

File: isc_libdemo.c [hunk 5]
```diff
@@ -61,58 +65,267 @@ int hello_world(struct m0_buf *in, struct m0_buf *out,
 		 */
 		out_str = m0_strdup("world");
 		if (out_str != NULL)
-			m0_buf_init(out, (void *)out_str, strlen(out_str));
+			m0_buf_init(out, out_str, strlen(out_str));
 		else
 			*rc = -ENOMEM;
```
Comment (nikitadanilov) [inline isc_libdemo.c:70]: *rc = M0_ERR(-ENOMEM), similarly in other places.

**Review by nikitadanilov — CHANGES_REQUESTED** [review]

File: isc/libdemo.h (hunk 4 above)
Comment (andriytk) [reply isc/libdemo.h:57]: Dropped as not used.

File: Makefile (hunk 1 above)
Comment (andriytk) [reply Makefile:196]: Thanks, fixed.

File: c0isc_demo.c (hunk 2 above)
Comment (andriytk) [reply c0isc_demo.c:280]: It's convenient to comment out / uncomment quickly with such style of comments. I know it's non-standard one, probably a gcc extension. But aren't we using a lot of gcc extensions anyway? So why not to use such style of comments?

File: isc/libdemo.h (hunk 3 above)
Comment (andriytk) [reply isc/libdemo.h:34]: Good question! Fixed.

File: isc_libdemo.c (hunk 5 above)
Comment (andriytk) [reply isc_libdemo.c:70]: m0tracing is not working for loadable .so modules and this won't be seen in m0traces anyway, so I don't see very much point in this.

Comment (nikitadanilov) [reply isc_libdemo.c:70]: Well, hopefully, we fix this problem, so it will be usable then.

**Review by nikitadanilov — CHANGES_REQUESTED** [review]

File: isc_libdemo.c (hunk 5 above)
Comment (andriytk) [reply isc_libdemo.c:70]: We might need some insights from @chumakd about this.

**Comment by andriytk** [comment]
Demo app and library was [moved](https://github.com/Seagate/cortx-motr/pull/857) to Motr repo, so closing this one.

**Pull request closed without merging**